### PR TITLE
add float8 types to dtypes table

### PR DESCRIPTION
**Pull request by vkuzo**
Summary:

As titled

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: docs/source/tensors.rst [hunk 1]
```diff
@@ -37,6 +37,8 @@ quantized 8-bit integer (unsigned)      ``torch.quint8``
 quantized 8-bit integer (signed)        ``torch.qint8``
 quantized 32-bit integer (signed)       ``torch.qint32``
 quantized 4-bit integer (unsigned) [3]_ ``torch.quint4x2``
+8-bit floating point, e4m3 [5]_         ``torch.float8_e4m3fn`` (limited support)
+8-bit floating point, e5m2 [5]_         ``torch.float8_e5m2`` (limited support)
```
Comment (drisspg) [inline docs/source/tensors.rst:41]: can we also add the 'nz' variants

File: docs/source/tensors.rst [hunk 2]
```diff
@@ -54,6 +56,11 @@ quantized 4-bit integer (unsigned) [3]_ ``torch.quint4x2``
   torch.compile); if you need eager support and the extra range is not needed,
   we recommend using their signed variants instead.  See
   https://github.com/pytorch/pytorch/issues/58734 for more details.
+.. [5]
+  `torch.float8_e4m3fn` and `torch.float8_e5m2` implement the spec for 8-bit
+  floating point types from https://arxiv.org/abs/2209.05433. The op support
```
Comment (drisspg) [inline docs/source/tensors.rst:61]: @albanD do you think we should link out to float8_experimental?

File: docs/source/tensors.rst (hunk 1 above)
Comment (vkuzo) [reply docs/source/tensors.rst:41]: the variants above are further along in their lifecycle as we actually have e2e things working throughout the stack and it's crossing the threshold of "this is useful".  I don't think the nz variants are quite there yet.

Comment (drisspg) [reply docs/source/tensors.rst:41]: @jeffdaily Is that a fair characterization, that we don't want them documented here

**Review by ezyang — APPROVED** [review]

File: docs/source/tensors.rst (hunk 1 above)
Comment (vkuzo) [reply docs/source/tensors.rst:41]: I'm going to land this, we can always follow up separately on the nz types

Comment (jeffdaily) [reply docs/source/tensors.rst:41]: AFAIK pytorch today doesn't have anything implemented to demonstrate the fnuz types on hardware.  We can follow up separately, even after #115214, once there might be some unit tests or other e2e demonstration.

OTOH, #115214 adds limited op support in pytorch for the fnuz types.

**Pull request closed without merging**